src/main/java/net/sacredlabyrinth/phaed/simpleclans/hooks/DiscordHook.java

```diff
@@ -334,32 +334,42 @@ public boolean categoryExists(String categoryId) {
     /**
      * Creates a new {@link ClanPlayer.Channel} in defined {@link Category}
      *
-     * <p>Sets {@link Permission#VIEW_CHANNEL} permission to all linked clan members.</p>
+     * <p>Sets positive {@link Permission#VIEW_CHANNEL} permission to all linked clan members.</p>
      *
-     * @return true, if channel was created. False, if category reached the limit.
+     * @throws InvalidChannelException clan is not verified or permanent, no one member is linked or clan is not in the whitelist.
+     * @throws LimitReachedException   if category reached the limit.
      */
-    public boolean createChannel(@NotNull Category category, @NotNull String clanTag) {
+    public void createChannel(@NotNull Category category, @NotNull String clanTag) throws InvalidChannelException, LimitReachedException {
```

Comment:
This should check if a channel for this clan already exists.

Reply:
Most likely, useless checking: it's not possible to catch this error at least *internally*.
Before creating channels, we [delete](https://github.com/RoinujNosde/SimpleClans/blob/4c881f4601812c9c7c38c6a6d553fd19b63f8400/src/main/java/net/sacredlabyrinth/phaed/simpleclans/hooks/DiscordHook.java#L454) those that already exist in discord.

Reply:
API users could try to create a channel that already exists.